lib/metadataTypes/EmailSendDefinition.js

```diff
@@ -110,7 +114,7 @@ class EmailSendDefinition extends MetadataType {
             // * this ignores r__assetMessage_Name on purpose as that is only unique per parent folder but useful during PR reviews
             metadata.Email.ID = cache.searchForField(
                 'asset',
-                metadata.r__assetMessage_Key,
+                metadata.r__assetMessage_Key.slice(0, Math.max(0, 36 - suffix.length)) + suffix,
```

Comment:
try catch  around this and in the catch part you try cache.searchForField() again but without your new logic. if that also fails, so be it and we show the error that we already defined.

Reply:
make sure to include the 2 delete statements in the try part, and in the catch part, after the cache.searchForField() part